lib/uuid.c

```diff
@@ -89,8 +89,15 @@ M0_INTERNAL int m0_uuid_parse(const char *str, struct m0_uint128 *val)
 	    parse_hex(&str[14],  4, &h3) < 0 || str[18] != '-' ||
 	    parse_hex(&str[19],  4, &h4) < 0 || str[23] != '-' ||
 	    parse_hex(&str[24], 12, &h5) < 0 || str[36] != '\0')
-		return M0_ERR(-EINVAL);
-	val->u_hi = h1 << 32 | h2 << 16 | h3;
+	    
+        if (parse_hex(&str[0],   8, &h1) < 0 ||
+            parse_hex(&str[8],   4, &h2) < 0 ||
+            parse_hex(&str[12],  4, &h3) < 0 ||
+            parse_hex(&str[16],  4, &h4) < 0 ||
+            parse_hex(&str[20], 12, &h5) < 0 || str[32] != '\0')
+                return M0_ERR(-EINVAL);
+
+    val->u_hi = h1 << 32 | h2 << 16 | h3;
```

Comment:
Please, ensure tabs and spaces are used in the right way. This patch introduces trailing white-spaces and spaces-instead-of-tab issues.

Reply:
Thanks for pointing this out I have updated the same in recent commit